### PR TITLE
Feat: block gas limit by default if not provided in rpc_sendTransaction

### PR DESCRIPTION
**Pull request by canonbrother**
## Summary

- as per title (it was a long time ago "TODO")

## Implications

- Storage
  - [ ] Database reindex required
  - [ ] Database reindex optional
  - [ ] Database reindex not required
  - [x] None

- Consensus
  - [ ] Network upgrade required
  - [ ] Includes backward compatible changes
  - [ ] Includes consensus workarounds
  - [ ] Includes consensus refactors
  - [x] None

### CODE REVIEW

File: lib/ain-rs-exports/src/evm.rs [hunk 1]
```diff
@@ -456,11 +456,17 @@ pub fn evm_unsafe_try_construct_block_in_q(
                 total_priority_fees,
                 block_number,
             }) => {
-                let Ok(total_burnt_fees) = u64::try_from(WeiAmount(total_burnt_fees).to_satoshi()) else {
```
Comment (sieniven) [inline lib/ain-rs-exports/src/evm.rs:459]: is there a reason for separating the else into a separate line?

File: lib/ain-rs-exports/src/evm.rs [hunk 2]
```diff
@@ -456,11 +456,17 @@ pub fn evm_unsafe_try_construct_block_in_q(
...
+                let Ok(total_burnt_fees) = u64::try_from(WeiAmount(total_burnt_fees).to_satoshi())
+                else {
                     return cross_boundary_error_return(result, "total burnt fees value overflow");
                 };
-                let Ok(total_priority_fees) = u64::try_from(WeiAmount(total_priority_fees).to_satoshi()) else {
```
Comment (sieniven) [inline lib/ain-rs-exports/src/evm.rs:462]: Same for this? Also the formatting is inconsistent with the above for the ease of reading (although clippy check passes)

File: lib/ain-rs-exports/src/evm.rs [hunk 3]
```diff
@@ -657,26 +663,38 @@ pub fn evm_try_get_tx_by_hash(
             let mut max_priority_fee_per_gas = 0u64;
             match &tx.transaction {
                 TransactionV2::Legacy(transaction) => {
-                    let Ok(price) = u64::try_from(WeiAmount(transaction.gas_price).to_satoshi()) else {
```
Comment (sieniven) [inline lib/ain-rs-exports/src/evm.rs:660]: Same as above

File: lib/ain-rs-exports/src/evm.rs [hunk 4]
```diff
@@ -657,26 +663,38 @@ pub fn evm_try_get_tx_by_hash(
...
+                    let Ok(price) = u64::try_from(WeiAmount(transaction.gas_price).to_satoshi())
+                    else {
                         return cross_boundary_error_return(result, "tx gas price value overflow");
                     };
                     gas_price = price;
                 }
                 TransactionV2::EIP2930(transaction) => {
                     tx_type = 1u8;
-                    let Ok(price) = u64::try_from(WeiAmount(transaction.gas_price).to_satoshi()) else {
```
Comment (sieniven) [inline lib/ain-rs-exports/src/evm.rs:667]: Same as above.

File: lib/ain-rs-exports/src/evm.rs [hunk 5]
```diff
@@ -657,26 +663,38 @@ pub fn evm_try_get_tx_by_hash(
...
+                    let Ok(price) = u64::try_from(WeiAmount(transaction.gas_price).to_satoshi())
+                    else {
                         return cross_boundary_error_return(result, "tx gas price value overflow");
                     };
                     gas_price = price;
                 }
                 TransactionV2::EIP1559(transaction) => {
                     tx_type = 2u8;
-                    let Ok(price) = u64::try_from(WeiAmount(transaction.max_fee_per_gas).to_satoshi()) else {
```
Comment (sieniven) [inline lib/ain-rs-exports/src/evm.rs:674]: Same as above.

File: lib/ain-rs-exports/src/evm.rs [hunk 6]
```diff
@@ -657,26 +663,38 @@ pub fn evm_try_get_tx_by_hash(
...
+                    let Ok(price) =
+                        u64::try_from(WeiAmount(transaction.max_fee_per_gas).to_satoshi())
+                    else {
+                        return cross_boundary_error_return(
+                            result,
+                            "tx max fee per gas value overflow",
+                        );
                     };
                     max_fee_per_gas = price;
-                    let Ok(price) = u64::try_from(WeiAmount(transaction.max_priority_fee_per_gas).to_satoshi()) else {
```
Comment (sieniven) [inline lib/ain-rs-exports/src/evm.rs:678]: Same as above.

File: lib/ain-rs-exports/src/evm.rs (hunk 1 above)
Comment (canonbrother) [reply lib/ain-rs-exports/src/evm.rs:459]: sigh.. its `fmt`
https://github.com/DeFiCh/ain/pull/2321/commits/a7ec9f5673089b52f3a203434e96a373cfd9db8f

File: lib/ain-rs-exports/src/evm.rs (hunk 2 above)
Comment (canonbrother) [reply lib/ain-rs-exports/src/evm.rs:462]: hmm.. is your fmt good which didn't trigger this alert??
lemme drop this.. just with `--no-verify`

**Review by sieniven — APPROVED** [review]

**Comment by shohamc1** [comment]
In geth, the default gas limit is set to 90000([ref](https://github.com/ethereum/go-ethereum/blob/e61035c5a3630e4f6fd0fb3e5346a4eed8cedc80/internal/ethapi/api.go#L50)). Wonder if we should adopt the geth standard or stick to this?

**Comment by canonbrother** [comment]
> In geth, the default gas limit is set to 90000([ref](https://github.com/ethereum/go-ethereum/blob/e61035c5a3630e4f6fd0fb3e5346a4eed8cedc80/internal/ethapi/api.go#L50)). Wonder if we should adopt the geth standard or stick to this?

hmm... default by block gas limit is more user friendly tho

**Comment by prasannavl** [comment]
This isn't the correct attribute? This is the block limit. +1 for upping the default TX limit, but let's turn it configurable with a node flag.

**Comment by canonbrother** [comment]
> This isn't the c

**Comment by canonbrother** [comment]
> This isn't the correct attribute? This is the block limit. +1 for upping the default TX limit, but let's turn it configurable with a node flag.

gosh.. lemme grab the block gas limit from header

**Comment by canonbrother** [comment]
> This isn't the correct attribute? This is the block limit. +1 for upping the default TX limit, but let's turn it configurable with a node flag.

hey @prasannavl weird.. isn't it a block gas limit?
https://github.com/DeFiCh/ain/blob/938a56135f5cc4f55f1e35ffd07d5d22523bedee/src/ffi/ffiexports.h#L9
30 mil unit is correct

**Comment by canonbrother** [comment]
> gosh.. lemme grab the block gas limit from header

i can't do this yet before we have block gas limit computation

**Comment by canonbrother** [comment]
(dangling PR): redo if necessary

**Pull request closed without merging**